packages/cmrjs/tests/cmr-utils/test-cmr-utils.js

```diff
@@ -661,3 +676,40 @@ test('getFileDescription returns fallback if file name cannot be determined', (t
     'File to download'
   );
 });
+
+const testMetadataObjectFromCMRFile = (filename, etag = 'foo') => async (t) => {
+  // Simulate throwing a PreconditionFailed error from getObject() because
+  // LocalStack ignores the `IfMatch` (and the `IfNoneMatch`) param passed
+  // to S3.getObject()
+  const errorSelector = {
+    code: 'PreconditionFailed',
+    errorCode: 412,
+    message: 'At least one of the pre-conditions you specified did not hold'
+  };
+  const { metadataObjectFromCMRFile } = proxyquire('../../cmr-utils', {
+    '@cumulus/aws-client/utils': {
+      // Don't bother with retries
+      retryOnMissingObjectError: (fn) => fn
+    },
+    '@cumulus/aws-client/S3': {
+      getObject: (_, { IfMatch: actualEtag }) => {
+        t.is(actualEtag, etag);
+        throw Object.assign(new Error(), errorSelector);
+      }
+    }
+  });
+
+  const error = await t.throwsAsync(metadataObjectFromCMRFile(filename, etag));
+
+  t.like(error, errorSelector);
+};
+
+test(
+  'metadataObjectFromCMRFile throws PreconditionFailed when ETag does not match CMR XML metadata file',
+  testMetadataObjectFromCMRFile('s3://bucket/fake.cmr.xml')
+);
+
+test(
+  'metadataObjectFromCMRFile throws PreconditionFailed when ETag does not match CMR UMMG JSON metadata file',
+  testMetadataObjectFromCMRFile('s3://bucket/fake.cmr.json')
```

Comment:
Given the test mocking - is there any difference between this and the previous test? It's not a huge deal, just wondering

Reply:
They take 2 different paths to finally get to the call to `getObject`, so this tests both paths.